### ISSUE TITLE
timeline-vis dates default to 20th century

### ISSUE THREAD

**Issue by timwilson** [Darakah/obsidian-timelines#16]
Thanks for this cool plugin. I'm looking forward to using it extensively!

I created three timeline events ranging from A.D. 46 to A.D. 57 for testing purposes, and they're displaying perfectly in the standard vertical timeline view. When I try a `timeline-vis` view, however, it defaults to showing dates in the 1900s (from 1960–1960 instead of 40–60) where there's no data to display.

<img width="766" alt="Screen Shot 2021-04-26 at 7 51 40 AM" src="https://user-images.githubusercontent.com/345022/116085319-52bd2500-a664-11eb-91c4-9d70f6c5b7a9.png">

Here's my timeline data:

```
<span
	class='ob-timelines'
	data-date='46-00-00-00'
	data-title='Pauls first missionary journey'
	data-class='orange'
	data-type='range'
	date-end='48-00-00-00'>
</span>

<span
	class="ob-timelines"
	data-date="49-00-00-00"
	data-title="Paul's second missionary journey"
	data-class="orange"
	data-type="range"
	date-end="52-00-00-00">
</span>

<span
	class="ob-timelines"
	data-date="53-00-00-00"
	data-title="Paul's third missionary journey"
	data-class="orange"
	data-type="range"
	date-end="57-00-00-00">
</span>
```

And here's the `timeline-vis` section. It's not clear what elements are required, but I haven't found any combination including `minDate` and `maxDate` that makes any difference.

```timeline-vis
startDate=40
endDate=60
```

Any suggestions?

**Comment by Darakah** [comment]
@timwilson ,

Glad you enjoyed the plugin, if it was useful consider giving the Git Repo a STAR!

Now, to your problem. There are a couple of problems. This is what your input should be:

```html
<span
	class='ob-timelines'
	data-date='0046-01-01-00'
	data-title='Pauls first missionary journey'
	data-class='orange'
	data-type='range'
	data-end='0048-01-01-00'>
</span>

<span
	class="ob-timelines"
	data-date="0049-01-01-00"
	data-title="Paul's second missionary journey"
	data-class="orange"
	data-type="range"
	data-end="0052-01-01-00">
</span>

<span
	class="ob-timelines"
	data-date="0053-01-01-00"
	data-title="Paul's third missionary journey"
	data-class="orange"
	data-type="range"
	data-end="0057-01-01-00">
</span>
```
----

### The problems:
1. Unlike the vertical timeline which doesn't require valid dates, the horizantal timeline requires valid date format. In your input you are specifying day `00` and month `00` which do not exist and thus the plugin is ignoring the events as invalid.
2. To specify the year you want you need to write: `0048` and not `48` 
3. you are writing `datE-end` and not `datA-end`, and since the plugin expects a start and end since you are specifying the type as range. Not having an end range leads to the event being ignored.
4. Same for the `startDate` and `endDate`, they need to be 40 & 60. 

You don't need `data-class` since you are not defining it.

<img width="872" alt="Screen Shot 2021-04-26 at 4 09 06 PM" src="https://user-images.githubusercontent.com/29651407/116097653-9fc3ea00-a6aa-11eb-97f1-e3285cc90d48.png">

For future uses the `data-type` can be one of the following:
- range
- background
- point
- box

**Comment by timwilson** [comment]
Thanks for the clarification. I believe I have the data formatted correctly now, but I'm still doing something wrong with the `timeline-vis` portion. It's still rendering a timeline from 1960–1960 like my original screenshot. Here's what I have:

````
```timeline-vis
startDate=0040
endDate=0060
```
````

And here's my fixed timeline data:

```
<span
	class="ob-timelines"
	data-date="0046-01-01-00"
	data-title="Paul’s first missionary journey"
	data-type="range"
	data-end="0048-01-01-00">
</span>

<span
	class="ob-timelines"
	data-date="0049-01-01-00"
	data-title="Paul’s second missionary journey"
	data-type="range"
	data-end="0052-01-01-00">
</span>

<span
	class="ob-timelines"
	data-date="0053-01-01-00"
	data-title="Paul’s third missionary journey"
	data-type="range"
	data-end="0057-01-01-00">
</span>
```

**Comment by Darakah** [comment]
- Is the note in which you have those spans have the tag of the timeline? i.e. `#timeline`
- Do the events show up properly but the centering is wrong?
- Try putting this inside the `timeline-vis` block
```
startDate=-0001,01,01
endDate=0100,01,01
```

Seems like the centering year is not converted properly for years below 100. Will patch it in the next release.

**Comment by timwilson** [comment]
@Darakah, the first two bullets don't do anything, but your suggestion in the third bullet works. Looks like you're on to something about years < 100.

**Comment by Darakah** [comment]
Can you share the md note that you are using?

**Comment by timwilson** [comment]
````
---
tags: timeline, test
---
```timeline-vis
tags=test
startDate=-0001,01,01
endDate=100,01,01
```

<span
	class="ob-timelines"
	data-date="0046-01-01-00"
	data-title="Paul’s first missionary journey"
	data-type="range"
	data-end="0048-01-01-00">
</span>

<span
	class="ob-timelines"
	data-date="0049-01-01-00"
	data-title="Paul’s second missionary journey"
	data-type="range"
	data-end="0052-01-01-00">
</span>

<span
	class="ob-timelines"
	data-date="0053-01-01-00"
	data-title="Paul’s third missionary journey"
	data-type="range"
	data-end="0057-01-01-00">
</span>
````

Here's what it looks like:

<img width="673" alt="Screen Shot 2021-04-26 at 3 36 52 PM" src="https://user-images.githubusercontent.com/345022/116147442-5755fd80-a6a5-11eb-9634-3c7cffa03e74.png">

**Comment by Darakah** [comment]
Yes, that is correct. 
The only problem is that I need to fix the centering. 

What is wrong then?

**Comment by timwilson** [comment]
The only things seems to be with something like this:

````
```timeline-vis
tags=test
startDate=0040,01,01
endDate=0060,01,01
```
````

where it spans from 1940 to 1960 like I posted about before. Other than that I think it’s fine.

**Comment by Darakah** [comment]
Yes, that can be solved by simply not casting the date to an integer here 
https://github.com/Darakah/obsidian-timelines/blob/8d4172242a455d079586de664f629d64df21992e/src/utils.ts#L44
As it is removing the leading zeroes and for some reason javascript defaults to the 19s.
Will fix it in the next release.

**Comment by Parlazo** [comment]
had similar issues with special years ... may years should be seen and numbers and not years some how
how ever i got to year 0 with 0000-01-01-00 instead of using 0.

**Comment by OogieM** [comment]
Maybe related, I'm trying to make a timeline that ignores years. It's to document when I need to plant seeds in the garden or start seeds and is based on my local area frost date. I've tried using a 0000 year  and then the month and day but all the items stack up and no span. If I put in a year it all works. But I effectively want this to be aperpetual calendar not tied to a year. 

![Screen Shot 2021-08-18 at 5 50 22 AM](https://user-images.githubusercontent.com/2394446/129893381-12d86a27-7e67-475f-ae5a-16ca4494c8b7.png)

Code in the plant notes. 

```---
tags: timeline, test
---
```

```html
<span 
      class='ob-timelines' 
      data-date='0000-07-02-00' 
      data-title='Plant carrots' 
     data-class='yellow'
      data-type='range' 
      data-end="0000-08-31-00"> 
</span>
```

```---
tags: timeline, test
---
```
```html
<span 
      class='ob-timelines' 
      data-date='0000-08-01-00' 
      data-title='Plant lettuce' 
      data-type='range' 
      data-end="0000-08-31-00"> 
</span>
```

side question, I'm not getting any different colors on the items but that is secondary to getting the data to display properly first.
